temporalio/worker/_workflow_instance.py

```diff
@@ -414,6 +419,100 @@ def _apply_cancel_workflow(
+        async def run_update() -> None:
+            command = self._add_command()
+            command.update_response.protocol_instance_id = job.protocol_instance_id
+            try:
+                defn = self._updates.get(job.name) or self._updates.get(None)
+                if not defn:
+                    known_updates = sorted([k for k in self._updates.keys() if k])
+                    raise RuntimeError(
+                        f"Update handler for '{job.name}' expected but not found, and there is no dynamic handler. "
+                        f"known updates: [{' '.join(known_updates)}]"
+                    )
+                args = self._process_handler_args(
+                    job.name,
+                    job.input,
+                    defn.name,
+                    defn.arg_types,
+                    defn.dynamic_vararg,
+                )
+                handler_input = HandleUpdateInput(
+                    id=job.id,
+                    update=job.name,
+                    args=args,
+                    headers=job.headers,
+                )
+
+                # Always run the validator interceptor, which will only actually run a validator if one is defined.
+                with self._as_read_only():
+                    await self._inbound.handle_update_validator(handler_input)
+
+                # Accept the update
+                command.update_response.accepted.SetInParent()
+                command = None  # type: ignore
+
+                # Run the handler
+                success = await self._inbound.handle_update_handler(handler_input)
+                result_payloads = self._payload_converter.to_payloads([success])
+                if len(result_payloads) != 1:
+                    raise ValueError(
+                        f"Expected 1 result payload, got {len(result_payloads)}"
+                    )
+                command = self._add_command()
+                command.update_response.protocol_instance_id = job.protocol_instance_id
+                command.update_response.completed.CopyFrom(result_payloads[0])
+            except (Exception, asyncio.CancelledError) as err:
+                logger.debug(
+                    f"Update raised failure with run ID {self._info.run_id}",
+                    exc_info=True,
+                )
+                # All asyncio cancelled errors become Temporal cancelled errors
+                if isinstance(err, asyncio.CancelledError):
+                    err = temporalio.exceptions.CancelledError(
+                        f"Cancellation raised within update {err}"
+                    )
+                # Read-only issues during validation should fail the task
+                if isinstance(err, temporalio.workflow.ReadOnlyContextError):
+                    self._current_activation_error = err
+                    return
+                # All other errors fail the update
```

Comment:
I don't think this is true _during_ the update handler. I think we only fail the update on Temporal failure exceptions during the handler or fail the task otherwise (but yes, all exceptions during validator short of read-only-with-commands-ones we fail the update).

Reply:
Yes, I had the same comment here.

Reply:
Fixed